source/common/router/rds_impl.cc

```diff
@@ -293,9 +290,9 @@ void RdsRouteConfigProviderImpl::onConfigUpdate() {
 }
 
 void RdsRouteConfigProviderImpl::validateConfig(
-    const envoy::config::route::v3::RouteConfiguration& config) const {
-  // TODO(lizan): consider cache the config here until onConfigUpdate.
-  ConfigImpl validation_config(config, optional_http_filters_, factory_context_, validator_, false);
+    const envoy::config::route::v3::RouteConfiguration&) const {
+  // Validation work is delegated to the subscription, since subscription is the
+  // entity that generate new ConfigImpl.
```

Comment:
Will this change impact: https://github.com/envoyproxy/envoy/blob/bfecc0025bd4b195a2b240a24ecc3c7330c823ec/source/common/router/rds_impl.cc#L126 ?
IIUC it will only perform the validation later (in the call to `onRdsUpdate()`) so I guess it shouldn't matter.
If so, should the call to the validation part be removed?